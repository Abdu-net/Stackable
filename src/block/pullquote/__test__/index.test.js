import { edit, save } from '../index'

describe( 'Pullquote', () => {
	test( 'block edit matches snapshots', () => {
		const wrapper = edit( {
			isSelected: false,
			attributes: {
				text: 'It\'s okay to acknowledge that life can get complicated, but we musn\'t forget the beauty in its simplicity, too. From the multitude of stars above, to freshly mowed grass in the morning, life is simply wonderful.',
				color: '#2091e1',
				borderColor: '#2091e1',
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )

	test( 'block save matches snapshots', () => {
		const wrapper = save( {
			isSelected: false,
			attributes: {
				text: 'It\'s okay to acknowledge that life can get complicated, but we musn\'t forget the beauty in its simplicity, too. From the multitude of stars above, to freshly mowed grass in the morning, life is simply wonderful.',
				color: '#2091e1',
				borderColor: '#2091e1',
			},

		} )

		expect( wrapper ).toMatchSnapshot()
	} )
} )
