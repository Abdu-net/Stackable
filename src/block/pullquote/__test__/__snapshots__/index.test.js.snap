// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pullquote block edit matches snapshots 1`] = `
<Fragment>
  <blockquote
    className="ugb-pullquote"
    style={
      Object {
        "--quote-color": undefined,
      }
    }
  >
    <RichText
      className="ugb-pullquote-text"
      formattingControls={
        Array [
          "bold",
          "italic",
          "strikethrough",
          "link",
        ]
      }
      isSelected={false}
      keepPlaceholderOnFocus={true}
      onChange={[Function]}
      placeholder="Write quoteâ€¦"
      style={
        Object {
          "color": "#2091e1",
        }
      }
      tagName="p"
      value="It's okay to acknowledge that life can get complicated, but we musn't forget the beauty in its simplicity, too. From the multitude of stars above, to freshly mowed grass in the morning, life is simply wonderful."
    />
  </blockquote>
  <InspectorControls>
    <PanelColorSettings
      colorSettings={
        Array [
          Object {
            "label": "Text Color",
            "onChange": [Function],
            "value": "#2091e1",
          },
          Object {
            "label": "Quote Color",
            "onChange": [Function],
            "value": undefined,
          },
        ]
      }
      title="Color Settings"
    />
  </InspectorControls>
</Fragment>
`;

exports[`Pullquote block save matches snapshots 1`] = `
<blockquote
  className="ugb-pullquote"
  style={
    Object {
      "--quote-color": undefined,
    }
  }
>
  <Unknown
    style={
      Object {
        "color": "#2091e1",
      }
    }
    tagName="p"
    value="It's okay to acknowledge that life can get complicated, but we musn't forget the beauty in its simplicity, too. From the multitude of stars above, to freshly mowed grass in the morning, life is simply wonderful."
  />
</blockquote>
`;
